src/main/scala/diplomacy/LazyModule.scala

```diff
@@ -144,61 +216,96 @@ sealed trait LazyModuleImpLike extends RawModule
   val auto: AutoBundle
   protected[diplomacy] val dangles: Seq[Dangle]
 
-  // .module had better not be accessed while LazyModules are still being built!
-  require (!LazyModule.scope.isDefined, s"${wrapper.name}.module was constructed before LazyModule() was run on ${LazyModule.scope.get.name}")
+  // [[wrapper.module]] had better not be accessed while LazyModules are still being built!
+  require (LazyModule.scope.isEmpty, s"${wrapper.name}.module was constructed before LazyModule() was run on ${LazyModule.scope.get.name}")
 
+  /** Set module name. */
   override def desiredName = wrapper.desiredName
   suggestName(wrapper.suggestedName)
 
+  /** [[Parameters]] for chisel [[Module]]s. */
   implicit val p = wrapper.p
 
+  /** [[instantiate]] will be called when a instance of [[LazyModuleImp]] is called. */
   protected[diplomacy] def instantiate() = {
     val childDangles = wrapper.children.reverse.flatMap { c =>
       implicit val sourceInfo = c.info
+      /** Calling [[c.module]] will push the real [[Module]] into [[chisel3.internal.Builder]].
+        * Note: this place is not calling [[instantiate]], it just push the [[c.module]] to Builder.
+        * */
       val mod = Module(c.module)
+      /** Ask each child to finish instantiate. */
       mod.finishInstantiate()
+      /** Return [[Dangle]]s of each child. */
       mod.dangles
     }
+
+    /** Ask each node in the [[LazyModule]] to call [[BaseNode.instantiate]].
+      * It will return a sequence of [[Dangle]] of these [[BaseNode]].
+      * */
```

Comment:
```suggestion

    // Ask each node in the [[LazyModule]] to call [[BaseNode.instantiate]].
    // It will return a sequence of [[Dangle]] of these [[BaseNode]].
```